Improve proposed changelog entry instructions (#7236)
jenkinsci/jenkins: .github/PULL_REQUEST_TEMPLATE.md

### Proposed changelog entries

<!--

> Please propose a changelog entry (or entries) for this pull request below between the lines with three backticks.

Reviewer: As an alternative, should we tell them to replace the specific text with their own text?
```suggestion
Please propose a changelog entry (or entries) for this pull request by replacing the 'Improve spline reticulation' text below with the changelog entry for the pull request.
```

Add more than one entry if necessary, but be aware that this is usually an indicator that a PR should be split into multiple PRs.
The changelog entry should be in the imperative mood; e.g., write "do this"/"return that" rather than "does this"/"returns that".

> Use basic HTML like '<strong>', '<a href="…">', '<code>' to format changelog entries. Please do not use Markdown or Asciidoc.

Reviewer 2: FWIW backticks are handled for code, https://github.com/jenkinsci/core-changelog-generator/blob/c13ec5065e4c16b836edc15d7d0a2b3ddd22f6ac/generate-jenkins-changelog.rb#L144

it wouldn't be hard to handle markdown links

For examples, see: https://www.jenkins.io/changelog/

Full changelog entry guideline: https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/_STYLEGUIDE.adoc#labels
-->

> ```
> - Improve spline reticulation.
> ```

Reviewer: The changelog generator works better when list items begin with `*` rather than `-`. List items that are started with `*` have worked the best in what I have seen.  When list items begin with `-`, the `-` is often included in the generated changelog message and needs to be removed by the docs SIG when reviewing and finalizing the changelog.

I'm not sure why we would put the proposed changelog entry in backticks.  Can you explain further why the backticks were added?  Is it intended that the pull request submitter will remove the backticks while creating the pull request?
```suggestion
* Improve spline reticulation.
```

Author: > Can you explain further why the backticks were added? Is it intended that the pull request submitter will remove the backticks while creating the pull request?

This prevents Markdown from rendering on the PR page, so it doesn't automatically look "good" in the PR (while being wrong for what we need).

Reviewer: I hadn't thought of that.  That's a great technique!  If someone mistakenly uses markdown or asciidoc, the backticks will increase the chance they will realize that they made a mistake.

Reviewer 2: any reason why you used 'Improve spline reticulation'?
```suggestion
```
- Replace this text with a proposed changelog entry.
```
```
